### DISCUSSION TITLE
2024.07.29の敵の生成のコミットからいろいろ

### DISCUSSION THREAD

**Discussion by AiletS** [CASru-GAME/TeamGameDevBootcamp#56] [General]
##  _の表記

今このプロジェクトでは変数名(主にprivateのとき)の一番最初に `_` をつけるときはその次の文字は小文字にすることになっています。

例 : `_button`

しかし、クラス名までこの表記を適用すると非常に不自然だと思います。

例 : `private BattleDebugEnemyGeneratePresenter _battleDebugEnemyGeneratePresenter`

クラス名は固有名詞みたいなものなのでクラス名に限っては `_` の後でも大文字にしませんか？

## フォルダーの構成

議論から`Debug`フォルダを別に作ってデバッグ昨日はそこで全部入れてしまおうっていうことになりましたが、これだと今の状態では名前空間の名前とスクリプトの名前が逆なので修正すべきだと思います。

例 : `App.Debug.Battle.Presenters` <=> `BattleDebugEnemyGeneratePresenter.cs`

- フォルダー構成を変える (デバッグ専用フォルダーを破棄して分ける)
- スクリプト(クラス)名を修正する `DebugBattleEnemyGeneratePresenter.cs`などに

しかし、後者だと既にあるクラス名を丸ごと含んでしまってわかりづらいので前者の方がいいと思います。

**Comment by momomo546** [comment]
参考にしてたやつだと、_を使ってなくてプライベートフィールドは小文字からになってましら。調べた感じだと、
_はプライベートであることを意味していて、
最初が小文字か大文字かというのと_は関係ないという感じでした。
なのでプライベートフィールドは_をつけて、名前はwikiの命名規則を参考にクラス、メソッド、パブリック変数などは大文字から、ローカル変数、パラメータ変数は小文字からがいいと思う。
